packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -145,65 +171,426 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
   return getGradleVersionFor(androidPluginVersion ?? 'unknown');
 }
 
-/// Returns true if [targetVersion] is within the range [min] and [max] inclusive.
-bool _isWithinVersionRange(
-  String targetVersion, {
-  required String min,
-  required String max,
-}) {
-  final Version? parsedTargetVersion = Version.parse(targetVersion);
-  final Version? minVersion = Version.parse(min);
-  final Version? maxVersion = Version.parse(max);
-  return minVersion != null &&
-      maxVersion != null &&
-      parsedTargetVersion != null &&
-      parsedTargetVersion >= minVersion &&
-      parsedTargetVersion <= maxVersion;
+/// Returns either the gradle-wrapper.properties value from the passed in
+/// [directory] or if not present the version available in local path.
+///
+/// If gradle version is not found null is returned.
+/// [directory] should be and android directory with a build.gradle file.
+Future<String?> getGradleVersion(
+    Directory directory, Logger logger, ProcessManager processManager) async {
+  final File propertiesFile = directory
+      .childDirectory(gradleDirectoryName)
+      .childDirectory(gradleWrapperDirectoryName)
+      .childFile(gradleWrapperPropertiesFilename);
+
+  if (propertiesFile.existsSync()) {
+    final String wrapperFileContent = propertiesFile.readAsStringSync();
+
+    // Expected content format (with lines above and below).
+    // Version can have 2 or 3 numbers.
+    // 'distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip'
+    final RegExp distributionUrlRegex =
+        RegExp(r'distributionUrl\s?=\s?.*\.zip');
```

Comment:
```suggestion
        RegExp(r'^\s*distributionUrl\s?=\s?.*\.zip');
```

That way this avoids matching a commented-out line.